numpy/_core/src/multiarray/arraytypes.c.src

```diff
@@ -2856,7 +2856,7 @@ OBJECT_nonzero (PyObject **ip, PyArrayObject *ap)
     }
     else {
         PyObject *obj;
-        memcpy(&obj, ip, sizeof(obj));
+        memcpy(&obj, (void *)ip, sizeof(obj));
```

Comment:
Technically, it would be nicer to change top to `char **` (or `void **` I guess, but char is more common in NumPy, since it works well with strided loops).